src/ol/source/ogcTileUtil.js

```diff
@@ -184,6 +188,42 @@ export function getVectorTileUrlTemplate(
     }
   }
 
+  if (collections && collections.length) {
+    // According to conformance class
+    // http://www.opengis.net/spec/ogcapi-tiles-1/1.0/conf/collections-selection
+    // commata in the identifiers of the `collections` query parameter
+    // need to be URLEncoded, while the commata separating the identifiers
+    // should not.
+    let url;
+    let isAbsolute = true;
+    try {
+      url = new URL(tileUrlTemplate);
+    } catch (e) {
+      isAbsolute = false;
+      url = new URL(tileUrlTemplate, 'https://example.com');
+    }
+
+    if (url.pathname.split('/').includes('collections')) {
+      logError(
+        'The "collections" query parameter cannot be added to collection endpoints',
+      );
+    } else {
+      const encodedCollections = collections
+        .map((c) => encodeURIComponent(c))
+        .join(',');
+
+      url.searchParams.append('collections', encodedCollections);
+      let baseUrl;
+      if (isAbsolute) {
+        baseUrl = decodeURI(url.toString()).split('?')[0];
+      } else {
+        baseUrl = decodeURIComponent(url.pathname);
+      }
```

Comment:
And now that we always have a correct absolute URL, you can simplify this too:
```suggestion
      const baseUrl = url.origin + decodeURIComponent(url.pathname);
```